src/controllers/UserController.js

```diff
@@ -0,0 +1,77 @@
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import models from "../models";
+import { loginValidation, signupValidation } from "../validator/userValidation";
+import { onError, onSuccess } from "../utils/response";
+
+export class UserController {
+  // create user
+  static async createUser(req, res) {
+    // validate signup
+    const { error } = signupValidation(req.body);
+    if (error) return onError(res, 400, error.details[0].message);
+    // check user if is already an admin
+
+    const emailExist = await models.User.findOne({
+      where: { email: req.body.email },
+    });
+    if (emailExist) return onError(res, 400, "Email already exist");
+
+    // Hash passwords
+
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(req.body.password, salt);
+
+    try {
+      const user = await models.User.create({
+        name: req.body.name,
+        email: req.body.email,
+        password: hashedPassword,
+      });
+      const saveUser = await user.save();
+      return onSuccess(res, 201, "Admin Signup successfully", saveUser);
+    } catch (err) {
+      console.log(err);
+      return onError(res, 500, "Internal Server error");
+    }
+  }
+  static async loginUser(req, res) {
+    const { error } = loginValidation(req.body);
+    if (error) return onError(res, 400, error.details[0].message);
+    // check if is exists
+
+    const user = await models.User.findOne({
+      where: { email: req.body.email },
+    });
+    if (!user) return onError(res, 401, "Invalid Email or Password");
+
+    // check if password is correct
+    const validPassword = await bcrypt.compare(
+      req.body.password,
+      user.password
+    );
+    if (!validPassword) return onError(res, 401, "Invalid Email or Password");
+
+    // create a token
+
+    const token = jwt.sign({ user: user }, process.env.JWT_SECRET);
```

Comment:
Do not expose all of the user information on the token, you can see that even password is in the user object u are signing